packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -502,16 +503,59 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
     return encodeAbiParameters(parseAbiParameters("bytes, address"), [moduleSignature, moduleAddressToUse]);
   }
 
+  async #getPaymasterUserOp(
+    userOp: Partial<UserOperationStruct>,
+    paymasterServiceData: PaymasterUserOperationDto,
+  ): Promise<Partial<UserOperationStruct>> {
+    if (paymasterServiceData.mode === PaymasterMode.SPONSORED) {
+      return this.#getPaymasterAndDataFromPaymaster(userOp, paymasterServiceData);
+    } else if (paymasterServiceData.mode === PaymasterMode.ERC20) {
```

Comment:
The "buildTokenPaymasterUserOp" is not called anymore ?

Reply:
Oh Christ, nice catch. I wonder how tests didn't fail 🤔

Reply:
Tests did not fail because "buildTokenPaymasterUserOp" adds token approval user op in order to approve the paymaster to spend the ERC20 for the user but probably our account that we use for testing already gave max approval to the paymaster.

Reply:
Nice catch. Could have been a nasty bug

Reply:
Updated